src/commonMain/kotlin/com/charleskorn/kaml/YamlInput.kt

```diff
@@ -115,6 +128,8 @@ public sealed class YamlInput(
         private fun YamlMap.withoutKey(key: String): YamlMap {
             return this.copy(entries = entries.filterKeys { it.content != key })
         }
+
+        private val SerialDescriptor.isContentBasedPolymorphic get() = serialName.startsWith(YamlContentPolymorphicSerializer::class.simpleName!!)
```

Comment:
Rather than checking the name like this, what if we introduced an annotation that is added to the `SerialDescriptor` created by `YamlContentPolymorphicSerializer`?

Reply:
The annotation is implemented as `internal` for now. Or should it be `public`?

Reply:
`internal` seems fine to me, it's an implementation detail.